### ISSUE TITLE
[storage::qnap::snmp::plugin] --mode=memory: wrong memory values returned

### ISSUE THREAD

**Issue by joschi99** [centreon/centreon-plugins#3185]
plugins 20210915 and latest plugins from github returns wrong memory values on different QNAP systems:

```
./centreon_plugins.pl --hostname=x.x.x.x --snmp-community=public --snmp-version=2c --plugin=storage::qnap::snmp::plugin --mode=memory  --warning-memory-usage-prct=80  --critical-memory-usage-prct=90 --warning-memory-usage= --warning-memory-usage-free= --critical-memory-usage= --critical-memory-usage-free=
CRITICAL: Memory total: 988.89 MB used: 988.31 MB (99.94%) free: 602.94 KB (0.06%) | 'memory.usage.bytes'=1036313660B;;;0;1036931072 'memory.free.bytes'=617412B;;;0;1036931072 'memory.usage.percentage'=99.94%;0:80;0:90;0;100
```

 snmpwalk .1.3.6.1.4.1.24681.1.2.2
`.1.3.6.1.4.1.24681.1.2.2.0 = STRING: "988.9 MB"`
snmpwalk .1.3.6.1.4.1.24681.1.2.3
`.1.3.6.1.4.1.24681.1.2.3.0 = STRING: "600.7 MB"`

snmpwalk .1.3.6.1.4.1.24681.1.3.2
`.1.3.6.1.4.1.24681.1.3.2.0 = Counter64: 1036931072`
 snmpwalk .1.3.6.1.4.1.24681.1.3.3
`.1.3.6.1.4.1.24681.1.3.3.0 = Counter64: 615584`

As you can see device is responding with legacy and ex values, but only legacy values are correct.

**Comment by garnier-quentin** [comment]
It's a known issue with qnap. Qnap will fix it.

**Comment by joschi99** [comment]
hmmm, but with the old plugin there was no problem on memory usage monitoring. And the are using FW versions like 4.3.4 or 4.2.6
As you can see legacy values are correct, only EX values are wrong. Maybe could be interest do force legacy with a dedicated option for this cases?

**Comment by garnier-quentin** [comment]
talked here: https://github.com/centreon/centreon-plugins/issues/2860
And here, we have removed: https://github.com/centreon/centreon-plugins/pull/3024

It's fixed in Qnap QTS 4.5.4.1741.

**Comment by garnier-quentin** [comment]
Add in last version
